### DISCUSSION TITLE
PHPDriver is deprecated, but does that mean that we are forced to put entities and their metadata together?

### DISCUSSION THREAD

**Discussion by MetalArend** [doctrine/doctrine2#10600] [General]
Working on a big project, where we try to keep our domain and infrastructure layers separated. We also want to avoid having to use xml files to keep that separation, so I went to see if there was an option to use php as standin replacement for the xml configuration. Turns out on https://www.doctrine-project.org/projects/doctrine-orm/en/2.14/reference/php-mapping.html#php-mapping it documents the php alternative for those xml files. But there's a big notice on the top stating that "PHPDriver is deprecated and will be removed in 3.0, use StaticPHPDriver instead."

Using the StaticPHPDriver is documented at https://www.doctrine-project.org/projects/doctrine-orm/en/2.14/reference/php-mapping.html#static-function, and it states "you want to keep your entity and mapping information together but don't want to use attributes or annotations". Well, ehm, we want to keep entity and mapping information separated in layers, so our domain can be used over multiple frameworks. Using attributes or annotations is already out of the question from the start, but are we forced to use xml (or heaven forbid yaml) to be able to do this? Or is there a php option I overlooked?

**Comment by greg0ire** [comment]
I do think XML is your only option here, unless maybe you want to write and maintain your own driver.

**Reply by MetalArend** [reply]
Maybe I should start writing more blogposts on how we keep big PHP projects maintainable/migratable, so use cases like ours would be more common. :rofl: Sometimes it feels like things are getting more and more difficult to do in a certain way, now that everyone seems to be embracing annotations and attributes.

Is that the reason the PHPDriver is getting deprecated, because of lack of maintainers? Is there such a big difference between what the StaticPHPDriver and the PHPDriver do behind the scenes?

**Reply by greg0ire** [reply]
Having more drivers does mean more maintenance yeah, and there are not a lot of us… maybe you could use a trait if you want to keep the separation.

Do write blog posts, it's always interesting to see other use cases :slightly_smiling_face:

**Reply by bernard-ng** [reply]
@MetalArend Hello, did you find a solution ? or switched to xml ?
